master/internal/api_workspace.go

```diff
@@ -551,11 +558,16 @@ func (a *apiServer) DeleteWorkspace(
 		return nil, err
 	}
 
+	modelsExist, _ := a.workspaceHasModels(ctx, req.Id)
+	if modelsExist {
+		return nil, fmt.Errorf("workspace (%d) contains models; move or delete models first", req.Id)
```

Comment:
Just returning an error like this will make this a 500 http error when it goes through the grpc-go gateway

If you instead return 
```
return status.Errorf(codes.FailedPrecondition, ...)
```

grpc gateway will make this a 400 error